src/codegen/mod.rs

```diff
@@ -1342,3 +1344,22 @@ impl From<&ast::Builtin> for Builtin {
         }
     }
 }
+
+pub(super) fn print_runtime_error(ns: &Namespace, error: &str, loc: Option<Loc>) -> String {
+    if let Some(loc) = loc {
+        match loc {
+            Loc::File(..) => {
+                let file_no = loc.file_no();
+                let curr_file = &ns.files[file_no];
+                let (line_no, offset) = curr_file.offset_to_line_column(loc.start());
+                format!("{} in {}:{}:{}", error, curr_file, line_no + 1, offset)
+            }
+            Loc::Codegen => {
+                format!("{error} in  codegen")
```

Comment:
Reporting this is a little strange. Every construct in codegen can be traced to something developers have types, except for the dispatch. A work for the future is to ensure the condegen expression are traced back to a location in sema.

Reply:
In that case, do you think only the error should be reported ?

Reply:
`codegen` won't mean anything to a regular user of the compiler. Reporting only the error would be the best option here. Lucas is right, ideally this should point out where in the source the problem is, we should fix this at some later point